### PR TITLE
(GH-190) Coding standards 

### PR DESCRIPTION
**Pull request by macfarmw**
Add contributing guide that outlines the naming rules and config files for ReSharper and EditorConfig support.

### CODE REVIEW

**Comment by ferventcoder** [comment]
This looks good. Can you clean up the commits so there is not a merge commit so we can merge? Thanks!

**Comment by macfarmw** [comment]
I'm happy to do that but I'm not sure I understand what you are asking for.

**Comment by macfarmw** [comment]
I thought I had it but I think I just ended up with the same thing again. I used rebase and squashed the last commits together but then I can't push that back to github unless I pull again first which adds back that merge commit.

**Comment by macfarmw** [comment]
OK. I think I have it. Is this right? I used git rebase -i to squash the recent commits and then git push --force to overwrite the remote repository.

File: roundhouse [hunk 1]
```diff
@@ -0,0 +1 @@
+Subproject commit 72c618d9f19ae7fe79d22f7a8fb37c32a2fc4077
```
Comment (ferventcoder) [inline roundhouse]: What is this? :D

**Comment by ferventcoder** [comment]
It all looks good, I just have a question about the roundhouse file that looks like a submodule.

**Comment by macfarmw** [comment]
Yikes! Here the heck did that come from? I'll have to fix that and squash again. It's OK I clearly need some practice on that anyway.

**Pull request closed without merging**